src/amo/components/RatingsByStar/index.js

```diff
@@ -0,0 +1,149 @@
+/* @flow */
+import makeClassName from 'classnames';
+import invariant from 'invariant';
+import * as React from 'react';
+import { connect } from 'react-redux';
+import { compose } from 'redux';
+
+import { fetchGroupedRatings } from 'amo/actions/reviews';
+import { withFixedErrorHandler } from 'core/errorHandler';
+import translate from 'core/i18n/translate';
+import LoadingText from 'ui/components/LoadingText';
+import Icon from 'ui/components/Icon';
+import type { GroupedRatingsType } from 'amo/api/reviews';
+import type { AppState } from 'amo/store';
+import type { AddonType } from 'core/types/addons';
+import type { ErrorHandlerType } from 'core/errorHandler';
+import type { I18nType } from 'core/types/i18n';
+import type { DispatchFunc } from 'core/types/redux';
+
+import './styles.scss';
+
+type Props = {|
+  addon: AddonType | null,
+|};
+
+type InternalProps = {|
+  ...Props,
+  dispatch: DispatchFunc,
+  errorHandler: ErrorHandlerType,
+  groupedRatings?: GroupedRatingsType,
+  i18n: I18nType,
+|};
+
+export class RatingsByStarBase extends React.Component<InternalProps> {
+  componentWillMount() {
+    // TODO: this is intended to load on the server (before mount) but it
+    // does not.
+    // See: https://github.com/mozilla/addons-frontend/issues/5854
+    this.loadDataIfNeeded(this.props);
+  }
+
+  componentWillReceiveProps(nextProps: InternalProps) {
+    this.loadDataIfNeeded(nextProps);
+  }
+
+  loadDataIfNeeded(props: InternalProps) {
+    const { addon, dispatch, errorHandler, groupedRatings } = props;
+    if (!errorHandler.hasError() && addon && !groupedRatings) {
+      dispatch(
+        fetchGroupedRatings({
+          addonId: addon.id,
+          errorHandlerId: errorHandler.id,
+        }),
+      );
+    }
+  }
+
+  renderBarValue(starCount: number) {
+    const { addon } = this.props;
+    invariant(addon, 'addon is required');
+
+    let width = 0;
+    if (addon.ratings && addon.ratings.count > 0) {
+      width = Math.round((starCount / addon.ratings.count) * 100);
+    }
+
+    return (
+      <div
+        className={makeClassName(
+          'RatingsByStar-bar',
+          'RatingsByStar-barValue',
+          {
+            'RatingsByStar-barValueLessThan100': width < 100,
```

Comment:
**OR**...maybe this class could be called RatingsByStar-fullBar - and flip styles for this new class

I don't know if this is any better actually but at first I thought this meant something else